platform/android/llapp_main.lua

```diff
@@ -15,6 +15,72 @@ end
 -- path to primary external storage partition
 local path = android.getExternalStoragePath()
 
+-- execute user scripts before patch.lua
+local lfs = require("libs/libkoreader-lfs")
+
+local user_dir = android.getExternalStoragePath() .. "/koreader"
+local afterupdate_marker = android.dir .. "/afterupdate.marker"
+local run_once_scripts = user_dir .. "/scripts.afterupdate"
+local run_always_scripts = user_dir .. "/scripts.always"
+
+local shell_scripts = {}
+
+local function getUserScripts(path)
+    shell_scripts = {} -- zero list, as getUserScripts could be called multiple times
+    local ret = lfs.attributes(path)
+    if ret == nil then
+        android.LOGI("no script folder " .. path .. " found!")
+        return
+    end
+    if ret.mode == "directory" then
+        android.LOGI("script folder " .. path .. " found!")
+        for entry in lfs.dir(path) do
```

Comment:
This should be the entry point of the getUserScripts function. Anything above is useless. If path is nil it will return no entries and the for loop won't execute

Reply:
I have removed the check.
Good to know, that lua is so smart.

Reply:
Well, is not technically correct what I said. I meant: the path is checked before calling getUserScript() for each one of the two paths. It is checked again on recursive calls (because it is only called if the entry is a directory). So there's no need to figure out if the path is a directory *inside* the function.

On functions **without** known arguments it is usually a good idea to check if function args are indeed not nil. This is usually done with:

```lua
function someFunction(arg)
    if not arg then return end
    (...)
end
```

Reply:
Remove lines 29 and 30.

Reply:
Done.